src/strands/multiagent/base.py

```diff
@@ -49,15 +52,48 @@ def get_agent_results(self) -> list[AgentResult]:
         """Get all AgentResult objects from this node, flattened if nested."""
         if isinstance(self.result, Exception):
             return []  # No agent results for exceptions
-        elif isinstance(self.result, AgentResult):
+        if isinstance(self.result, AgentResult):
             return [self.result]
-        else:
-            # Flatten nested results from MultiAgentResult
-            flattened = []
-            for nested_node_result in self.result.results.values():
-                flattened.extend(nested_node_result.get_agent_results())
+        if getattr(self.result, "__class__", None) and self.result.__class__.__name__ == "AgentResult":
+            return [self.result]  # type: ignore[list-item]
+
+        # If this is a nested MultiAgentResult, flatten children
+        if hasattr(self.result, "results") and isinstance(self.result.results, dict):
+            flattened: list[AgentResult] = []
+            for nested in self.result.results.values():
+                if isinstance(nested, NodeResult):
+                    flattened.extend(nested.get_agent_results())
             return flattened
 
+        return []
+
+    def to_dict(self) -> dict[str, Any]:
+        """Convert NodeResult to JSON-serializable dict, ignoring state field."""
+        result_data: Any = None
+        if isinstance(self.result, Exception):
+            result_data = {"type": "exception", "message": str(self.result)}
+        elif isinstance(self.result, AgentResult):
+            # Serialize AgentResult without state field
+            result_data = {
+                "type": "agent_result",
+                "stop_reason": self.result.stop_reason,
+                "message": self.result.message,  # Message type is JSON serializable
+                # Skip metrics and state - not JSON serializable
+            }
+        elif hasattr(self.result, "to_dict"):
```

Comment:
The type of `self.result` is `Union[AgentResult, "MultiAgentResult", Exception]` - can we handle all of those cases with `isinstance` checks?  I'm a bit concerned that we're introducing data that (A) we don't know the type of, and (B) is "loosely" typed instead of strongly typed